management_app/views/points.py

```diff
@@ -69,12 +67,31 @@ def calculate_teaching_point_val(course_title_id, num_of_enrollment, offload_or_
             y2 = year
             y3 = year
 
-        sql_statement = f"SELECT COUNT(*) FROM scheduled_teaching WHERE ((year = {y1} AND quarter = 1) OR (year = {y2} AND quarter = 2) OR (year = {y3} AND quarter = 3)) AND teaching_point_val = {point_c4}"
-        num_of_row = db.execute(sql_statement).fetchone()         
+        # count by "teaching_point_val" column rather than by "units" column (units = 2) because we only stop giving points when the faculty is already given 0.5 in total
+        sql_statement = f"SELECT teaching_point_val FROM scheduled_teaching st JOIN courses ON st.course_title_id = courses.course_title_id WHERE ((year = {y1} AND quarter = 1) OR (year = {y2} AND quarter = 2) OR (year = {y3} AND quarter = 3)) AND units = 2 AND user_id = {user_id}"
+        rows = db.execute(sql_statement).fetchall()
```

Comment:
Using format strings to create a SQL statement is vulnerable to SQL injection. Use `db.execute` to create the SQL statement.